src/libraries/Position.sol

```diff
@@ -26,20 +26,24 @@ library Position {
     /// @param owner The address of the position owner
     /// @param tickLower The lower tick boundary of the position
     /// @param tickUpper The upper tick boundary of the position
+    /// @param salt A unique value to differentiate between multiple positions in the same range
     /// @return position The position info struct of the given owners' position
-    function get(mapping(bytes32 => Info) storage self, address owner, int24 tickLower, int24 tickUpper)
+    function get(mapping(bytes32 => Info) storage self, address owner, int24 tickLower, int24 tickUpper, bytes32 salt)
         internal
         view
         returns (Info storage position)
     {
-        // positionKey = keccak256(abi.encodePacked(owner, tickLower, tickUpper))
+        // positionKey = keccak256(abi.encodePacked(owner, tickLower, tickUpper, salt))
         bytes32 positionKey;
+
         /// @solidity memory-safe-assembly
         assembly {
-            mstore(0x06, tickUpper) // [0x23, 0x26)
-            mstore(0x03, tickLower) // [0x20, 0x23)
-            mstore(0, owner) // [0x0c, 0x20)
-            positionKey := keccak256(0x0c, 0x1a)
+            let freeMemoryPointer := mload(0x40)
+            mstore(add(freeMemoryPointer, 0x26), salt)
+            mstore(add(freeMemoryPointer, 0x06), tickUpper)
+            mstore(add(freeMemoryPointer, 0x03), tickLower)
+            mstore(freeMemoryPointer, owner)
+            positionKey := keccak256(add(freeMemoryPointer, 0x0c), 0x3a) // len is 58 bytes
```

Comment:
```suggestion
            mstore(0x26, salt)
            mstore(0x06, tickUpper)
            mstore(0x03, tickLower)
            mstore(0, owner)
            positionKey := keccak256(0x0c, 0x3a) // len is 58 bytes
            mstore(0x26, 0)
```
The upper bits of the free memory pointer slot should always be zero and safely erased.

Reply:
If the upper 6 bytes is non-zero, `memory_size_word` is at least `1 << (256 - 48) / 32 = 1 << 203`. `memory_cost` would be greater than `1 << 397`. Gas would've run out by then.
https://www.evm.codes/about#memoryexpansion